src/util/io_util.rs

```diff
@@ -22,3 +23,25 @@ impl<R: Read> Read for LimitErrorReader<R> {
         }
     }
 }
+
+pub fn read_le_u32<R: Read + ?Sized>(r: &mut R) -> io::Result<u32> {
```

Comment:
I have no idea, probably! This code probably predates the byteorder crate :) I didn't change these functions at all, just moved them around, and if we change to use the byteorder crate, I'd want to do that in a separate PR.